index.js

```diff
@@ -387,6 +387,9 @@ class TuyaDevice extends EventEmitter {
 
     if (packet.commandByte === CommandType.HEART_BEAT) {
       debug(`Pong from ${this.device.ip}`);
+
+      // Remove resolver
+      delete this._resolvers[packet.sequenceN];
```

Comment:
Ah yes, but `HEART_BEAT` packets have not been assigned a `sequenceN`, so we're still deleting `this._resolvers[0]` every time.

Are you sure there are no cases when two `HEART_BEAT` packets would arrive? Could network lag, device failure, or a malicious actor create this situation?

What about giving each packet a unique `sequenceN`? No special treatment for `HEART_BEAT` packets.

To clarify, what do you think about [something along these lines](https://github.com/kueblc/tuyapi/commit/a3299cb7cd9ca006d0b4da0aa1a97d41b8ab917a)? I left the `HEART_BEAT` debug message, along with the early `return` so no `data` event is emitted for `HEART_BEAT`, though this could be moved too. I suppose most use cases shouldn't need to know about these anyway.

Reply:
I added your changes, makes sense.

The only issue left is that there's no data attached to the resolver for the `set()` calls now.

What do you think about having a separate resolver property for set commands?